libkbfs/block_journal.go

```diff
@@ -860,65 +860,107 @@ func (j *blockJournal) saveBlocksUntilNextMDFlush() error {
 	return nil
 }
 
-func (j *blockJournal) onMDFlush(ctx context.Context) error {
+// onMDFlush removes at most `maxToRemove` blocks from the
+// `saveUntilMDFlush` journal if one exists.  If `lastToRemove` is
+// zero, it flushes the complete `saveUntilMDFlush` if it has fewer
+// than `maxToRemove` entries; if it doesn't flush the entire journal,
+// it returns the ordinal of the current last entry.  If
+// `lastToRemove` is non-zero, it only flushes up to the minimum of
+// `lastToRemove` and the earliest entry + `maxToRemove`; if this
+// flushes the entire saved journal, it returns 0, otherwise it
+// returns `lastToRemove`.
+func (j *blockJournal) onMDFlush(ctx context.Context,
+	maxToRemove uint64, lastToRemove journalOrdinal) (journalOrdinal, error) {
 	if j.saveUntilMDFlush == nil {
-		return nil
+		return 0, nil
+	}
+
+	if maxToRemove == 0 {
+		return 0, errors.New("maxToRemove must be non-zero")
 	}
 
 	// Delete the block data for anything in the saved journal.
 	first, err := j.saveUntilMDFlush.readEarliestOrdinal()
 	if os.IsNotExist(err) {
-		return nil
+		return 0, nil
 	} else if err != nil {
-		return err
+		return 0, err
 	}
 
 	last, err := j.saveUntilMDFlush.readLatestOrdinal()
 	if os.IsNotExist(err) {
-		return nil
+		return 0, nil
 	} else if err != nil {
-		return err
+		return 0, err
 	}
 
-	j.log.CDebugf(ctx, "Removing saved data for entries [%d, %d]", first, last)
-	for i := first; i <= last; i++ {
+	if lastToRemove != 0 {
+		if last < lastToRemove {
+			return 0, fmt.Errorf("Last removal requested is %d, but "+
```

Comment:
Once I merge my PR that uses pkg/errors, please change this and others to `errors.Errorf`